core/src/main/java/org/infinispan/factories/threads/SecurityActions.java

```diff
@@ -15,18 +14,7 @@
  * @since 14.0
  */
 final class SecurityActions {
-   private static <T> T doPrivileged(PrivilegedAction<T> action) {
-      if (System.getSecurityManager() != null) {
-         return AccessController.doPrivileged(action);
-      } else {
-         return Security.doPrivileged(action);
-      }
-   }
-
    static void setContextClassLoader(Thread thread, ClassLoader contextClassLoader) {
-      doPrivileged(() -> {
-         thread.setContextClassLoader(contextClassLoader);
-         return null;
-      });
+      doPrivileged(new SetThreadContextClassLoaderAction(contextClassLoader));
```

Comment:
is `doPrivileged` still needed? I also noticed the `thread` is ignored and sets the classload in the current thread.